### PR TITLE
upgrade-db: add XL migration script

### PR DESCRIPTION
**Pull request by jean-edouard**
OXT-915
OXT-1059

Signed-off-by: Jed <lejosnej@ainfosec.com>
Signed-off-by: Troy Crosley <crosleyt@ainfosec.com>

### CODE REVIEW

**Comment by crogers1** [comment]
I went through a bunch of the xl commits and couldn't find any more config entries that changed, so this has a +1 from me.

**Comment by jean-edouard** [comment]
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/713

**Comment by dpsmith** [comment]
Has this been tested yet?

File: upgrade-db/Migrations/M_34.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+act :: IO ()
+act = do
+  updateGuests
+
+updateGuests = xformVmJSON xform where
+  xform tree = case jsGet "/type" tree of
+    Just s | jsUnboxString s == "svm" -> modify tree
+    _ -> tree
+    where
+      modify = jsSet "/v4v-firewall-rules/4" (jsBoxString "my-stubdom -> 0:5100") .
```
Comment (cjp256) [inline upgrade-db/Migrations/M_34.hs]: Is there a better option to append rather than hard-coding /4 entry?

**Comment by jean-edouard** [comment]
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/812

File: upgrade-db/Migrations/M_34.hs (hunk 1 above)
Comment (jean-edouard) [inline upgrade-db/Migrations/M_34.hs]: I'll try... See https://openxt.atlassian.net/browse/OXT-1088
In the meantime, I think this PR should still go in, to enable upgrade testing.
Thanks

**Comment by jean-edouard** [comment]
Fixed, improved, rebased and cleaned up here:
https://github.com/OpenXT/manager/pull/88

Closing soon.

**Pull request closed without merging**